internal/engine/eval/rego/datasources.go

```diff
@@ -0,0 +1,78 @@
+// SPDX-FileCopyrightText: Copyright 2023 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package rego
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/types"
+
+	v1datasources "github.com/mindersec/minder/pkg/datasources/v1"
+)
+
+// RegisterDataSources implements the Eval interface.
+func (e *Evaluator) RegisterDataSources(dsr *v1datasources.DataSourceRegistry) {
+	for key, dsf := range dsr.GetFuncs() {
+		fmt.Printf("Registering data source %s\n", key)
+		e.regoOpts = append(e.regoOpts, buildFromDataSource(key, dsf))
```

Comment:
Question: does it make sense for `dsf` from `dsr.GetFuncs()` to actually have a `func(*rego.Rego)` method (i.e. to move the `buildFromDataSource` into a function on `DataSourceFuncDef`)?

It pushes the Rego types into a slighly larger area of the code, but it helps avoid having a big ball of code in `eval/rego` (and if we name the method `Rego` or `RegoOpt`, we would have a pattern for any other future evaluation engines).

Reply:
I wanted to have data sources to be quite independent from the eval engine. The eval engine would then be in charge of interpreting the data source and making sense of it. It brings a lot more responsibility into the eval engine, but it allows for us to include more engines with ease in the future. e.g. if we'd introduce a WASM engine, we wouldn't need to mess around with rego structures.